src/lib/structures/uid.ml

```diff
@@ -26,38 +26,65 @@
 (**************************************************************************)
 
 module DStd = Dolmen.Std
-module DE = DStd.Expr
+module DE = Dolmen.Std.Expr
 
 type t =
   | Hstring : Hstring.t -> t
   | Dolmen : 'a DE.id -> t
+  | Term_cst : DE.term_cst -> t
+  | Ty_cst : DE.ty_cst -> t
 
 let[@inline always] of_dolmen id = Dolmen id
+let[@inline always] of_term_cst id = Term_cst id
```

Comment:
Is it correct to say that `of_term_cst` should only be called on term constructors that have had an order attached? Could we had an assertion for this here?

Reply:
It is not true because we use it also for unique constructor of record but I can generate an order for these constructors too and add an assertion here.

Reply:
Actually, we cannot put an assertion here as destructors are also of type `term_cst`.